libbpf-tools/biolatency.bpf.c

```diff
@@ -52,17 +57,42 @@ int trace_rq_start(struct request *rq)
 }
 
 SEC("tp_btf/block_rq_insert")
-int BPF_PROG(block_rq_insert, struct request_queue *q, struct request *rq)
+int block_rq_insert(u64 *ctx)
 {
-	return trace_rq_start(rq);
+	/**
+	 * commit a54895fa (v5.11-rc1) changed tracepoint argument list
+	 * from TP_PROTO(struct request_queue *q, struct request *rq)
+	 * to TP_PROTO(struct request *rq)
+	 */
+	if (LINUX_KERNEL_VERSION <= KERNEL_VERSION(5, 10, 0))
+		return trace_rq_start((void *)ctx[1]);
+	else
+		return trace_rq_start((void *)ctx[0]);
 }
 
-SEC("tp_btf/block_rq_issue")
-int BPF_PROG(block_rq_issue, struct request_queue *q, struct request *rq)
+static __always_inline
+int f(u64 *ctx, int idx)
```

Comment:
f() is not a good name, please come up with something more sane. And why passing ctx and idx separately, not just desired pointer type (`struct request *`)?  Makes no sense to do ctx[idx] everywhere, if you can do it down in block_rq_issue very explicitly.

Reply:
agree, reworked that part.